lib/core/src/wallet.rs

```diff
@@ -104,20 +106,47 @@ impl LiquidOnchainWallet {
         persister: Arc<Persister>,
         user_signer: Arc<Box<dyn Signer>>,
     ) -> Result<Self> {
-        let signer = crate::signer::SdkLwkSigner::new(user_signer)?;
-        let descriptor = LiquidOnchainWallet::get_descriptor(&signer, config.network)?;
-        let elements_network: ElementsNetwork = config.network.into();
+        let signer = crate::signer::SdkLwkSigner::new(user_signer.clone())?;
+        let wollet = Self::create_wallet(&config, working_dir, &signer)?;
 
-        let lwk_persister = FsPersister::new(working_dir, elements_network, &descriptor)?;
-        let wollet = Wollet::new(elements_network, lwk_persister, descriptor)?;
         Ok(Self {
             config,
             persister,
             wallet: Arc::new(Mutex::new(wollet)),
+            working_dir: working_dir.clone(),
             signer,
         })
     }
 
+    fn create_wallet<P: AsRef<Path>>(
+        config: &Config,
+        working_dir: P,
+        signer: &SdkLwkSigner,
+    ) -> Result<Wollet> {
+        let elements_network: ElementsNetwork = config.network.into();
+        let descriptor = LiquidOnchainWallet::get_descriptor(signer, config.network)?;
+        let mut lwk_persister =
+            FsPersister::new(working_dir.as_ref(), elements_network, &descriptor)?;
+
+        match Wollet::new(elements_network, lwk_persister, descriptor.clone()) {
+            Ok(wollet) => Ok(wollet),
+            Err(lwk_wollet::Error::UpdateHeightTooOld { .. }) => {
+                warn!("Update height too old, wipping storage and retrying");
+                let mut path = working_dir.as_ref().to_path_buf();
+                path.push(elements_network.as_str());
```

Comment:
I think the passed working_dir is already formatted with the network/fingerprint here: 
https://github.com/breez/breez-sdk-liquid/blob/8f3c303534f4f608afd4a1f38e96a82c5a1c1f68/lib/core/src/sdk.rs#L156
Do we need to add it again?

Reply:
Yes but the lwk wallet dir is a nested directory under that directory.

Reply:
Ah right, it's the ElementsNetwork, sorry